src/consensus/aft/raft.h

```diff
@@ -284,6 +282,32 @@ namespace aft
       return replica_state == kv::ReplicaState::Follower;
     }
 
+    bool is_eligible_voter(const ccf::NodeId& from) const
+    {
+      if (from == state->my_node_id)
+      {
+        return !catching_up && !retiring;
+      }
+      else
+      {
+        auto it = nodes.find(from);
+        return it != nodes.end() && !it->second.catching_up;
```

Comment:
A few questions, to help me understand the implementation:

Does the local state need to be different than the state for all the other nodes?
Do we not track whether other nodes are retiring?
Do we have an entry for `state->my_node_id` in `nodes`, and if so does `nodes[me].catching_up` always match `catching_up`?

Reply:
Not sure, yes, and no. I had tried not to disrupt the existing code too much (e.g. `me` not in `nodes`), but after the latest discussions I'm now changing this.